### PR TITLE
Database updates

### PR DESCRIPTION
**Pull request by Mastergatto**
By comparing with No-Intro database (see #599 ), I have corrected some errors according to No-Intro, because IMHO No-Intro is more trustworthy than GoodSet. I have also added new iQue games and manuals, some previously undumped revisions and some new prototypes.  

Advices are welcome.

### CODE REVIEW

File: data/mupen64plus.ini [hunk 1]
```diff
@@ -6200,17 +6312,22 @@ GoodName=Hiryuu no Ken Twin (J) [h3C]
...
+[FA6A2E00BBFF236DAC91F08544AFFA40]
 GoodName=Holy Magic Century (E) [b1]
 CRC=277B129D DD3879FF
-RefMD5=FA6A2E00BBFF236DAC91F08544AFFA40
+RefMD5=80CC112F62E9A8581A1BB6A1D1E1488B
+
+[FA6A2E00BBFF236DAC91F08544AFFA40]
+GoodName=Holy Magic Century (E) [b2]
+CRC=277B129D DD3879FF
+RefMD5=80CC112F62E9A8581A1BB6A1D1E1488B
```
Comment (Jj0YzL5nvJ) [inline data/mupen64plus.ini]: Remove this, it's wrong from beginning

File: data/mupen64plus.ini [hunk 2]
```diff
@@ -8823,40 +8989,39 @@ CRC=73C0403D E821D951
...
 GoodName=Mario Golf (E) [!]
 CRC=62E957D0 7FC15A5D
 Players=4
 SaveType=SRAM
 Rumble=Yes
 Transferpak=Yes
 
+[4D010AE1AF4B04D6B70B799C56F05993]
+GoodName=Mario Golf (E) [b1]
+RefMD5=55634FF90EE997790781F79A5B0097EE
```
Comment (Jj0YzL5nvJ) [inline data/mupen64plus.ini]: Is missing: "CRC=62E957D0 7FC15A5D"

File: data/mupen64plus.ini [hunk 3]
```diff
@@ -8867,39 +9032,40 @@ Transferpak=Yes
...
+RefMD5=903E6929666531D72D05D1E4C522E305
 
 [12569B4FA5239A3AD53DE5D2E0243488]
 GoodName=Mario Golf (U) [t1]
 CRC=26647A17 8D3980C3
-RefMD5=7A5D0D77A462B5A7C372FB19EFDE1A5F
+RefMD5=903E6929666531D72D05D1E4C522E305
 
-[E5A041B1B7C8E3B4C2E8178E5A138E2D]
+
```
Comment (Jj0YzL5nvJ) [inline data/mupen64plus.ini]: This is not necessary.

File: data/mupen64plus.ini [hunk 4]
```diff
@@ -10078,6 +10267,10 @@ GoodName=Mortal Kombat Trilogy (U) (V1.2) [t6] (P1 Invincible)
 CRC=83F33AA9 A901D40D
 RefMD5=3DCB15043063BD656A0223C519218CFB
 
+[B6A7B85C123DF4E0B9CC118578DBC9A2]
+GoodName=Mortal Kombat Trilogy (U) (Beta) (1996-05-13) [!]
```
Comment (Jj0YzL5nvJ) [inline data/mupen64plus.ini]: Not CRC?

File: data/mupen64plus.ini [hunk 5]
```diff
@@ -11816,6 +12042,13 @@ Players=1
 SaveType=Flash RAM
 CountPerOp=1
 
+[33FDAB9712D9FEA793A3AE44293999C3]
+GoodName=Pocket Monsters Snap (J) (VC) [!]
+CRC=E0044E9E CD659D0D
+Players=1
+SaveType=Flash RAM
+CountPerOp=1
```
Comment (Jj0YzL5nvJ) [inline data/mupen64plus.ini]: Replace with this with: RefMD5=FBDD74ED68E6A0CD734562D56CCB752D

File: data/mupen64plus.ini [hunk 6]
```diff
@@ -15907,27 +16208,35 @@ SaveType=None
...
 Mempak=Yes
 Rumble=Yes
 
+[C38ACBAE773CC3845EA354421E171998]
+GoodName=Turok 3 - Shadow of Oblivion (E) (Beta) (2000-07-10)
+CRC=D25C1211 13EEBF67
+Players=4
+SaveType=None
+Mempak=Yes
+Rumble=Yes
```
Comment (Jj0YzL5nvJ) [inline data/mupen64plus.ini]: Replace 'Players', 'SaveType', 'Mempak' and 'Rumble' for: RefMD5=279EC83BD60A3CCE69A1DB22B0A5C318

File: data/mupen64plus.ini [hunk 7]
```diff
@@ -15907,27 +16208,35 @@ SaveType=None
...
+
+[0A3A4C761960F7D648AFA60B1E565C7C]
+GoodName=Turok 3 - Shadow of Oblivion (U) (Beta) [h1C]
+CRC=D25C1211 13EEBF67
+RefMD5=69CE88C46A7C829C6F54004DE93EFCEF
```
Comment (Jj0YzL5nvJ) [inline data/mupen64plus.ini]: Replace with: RefMD5=279EC83BD60A3CCE69A1DB22B0A5C318

File: data/mupen64plus.ini (hunk 4 above)
Comment (Mastergatto) [reply data/mupen64plus.ini]: The emulator didn't return anything for this file, so no CRC.

**Comment by Mastergatto** [comment]
I have commited those changes as you have recommended, @Jj0YzL5nvJ . Is there anything other that can I do? I'll squash all changes into single commit if needed.

File: data/mupen64plus.ini (hunk 4 above)
Comment (Mastergatto) [reply data/mupen64plus.ini]: It's also the same for almost all iQue games and manuals.

Comment (Jj0YzL5nvJ) [reply data/mupen64plus.ini]: Even unsupported homebrew generate CRC's... like [Flappy Bird 64](https://github.com/meeq/FlappyBird-N64)

**Comment by Jj0YzL5nvJ** [comment]
With:
> Replace 'Players', 'SaveType', 'Mempak' and 'Rumble' for: RefMD5=279EC83BD60A3CCE69A1DB22B0A5C318

I was referring to all beta versions... but that's enough, this file has too many redundancies.

> Is there anything other that can I do? I'll squash all changes into single commit if needed.

Do that, that always pleases most devs.

**Comment by Mastergatto** [comment]
It should be all good now, I have even replaced these entries with RefMD5. As for these missing CRCs, someone should investigate why mupen64plus is unable to generate them for those ROMs.

**Comment by loganmc10** [comment]
@Jj0YzL5nvJ you're good with this? It seems like you reviewed it. If you're happy with I'd say we merge it

**Comment by Jj0YzL5nvJ** [comment]
I'm fine with this. But it is clear that the current DB is obsolete and has too many redundancies.
This needs a SHA1 and maybe others, differentiate between N64, Aleck64, 64DD, iQue, VC, etc. And differentiate the ROM and boot method / save based on such. The current method does not work with iQue.

The whole thing needs a new "M64P_INICreater tool" to revamp. Using GoodSet, No-Intro or / and any other, like [MAME's hash](https://github.com/mamedev/mame/tree/master/hash).

But without new blood, that won't happen. So let's hotfix this.

**Comment by fzurita** [comment]
@Narann this has a problem. Section 31448B07CBC932018870F71D1254E317 is repeated twice.

**Comment by Narann** [comment]
Merged!

**Pull request merged**